src/lib/stepper/stepper.ts

```diff
@@ -66,12 +67,17 @@ export class MatStep extends _MatStep implements ErrorStateMatcher {
 @Directive({
   selector: '[matStepper]'
 })
-export class MatStepper extends _MatStepper {
+export class MatStepper extends _MatStepper implements AfterContentInit {
   /** The list of step headers of the steps in the stepper. */
   @ViewChildren(MatStepHeader, {read: ElementRef}) _stepHeader: QueryList<ElementRef>;
 
   /** Steps that the stepper holds. */
   @ContentChildren(MatStep) _steps: QueryList<MatStep>;
+  
+  /** Workaround for https://github.com/angular/material2/issues/8397 */
+  ngAfterContentInit() {
+    this._steps.changes.pipe(takeUntil(this._destroyed)).subscribe(() => this._stateChanged());
+  }
```

Comment:
Maybe change the comment to be like this?

```ts
ngAfterContentInit() {
  // Mark the component for change detection whenever the content children query changes
  this._steps.changes.pipe(...).subscribe(...);
}
```

Reply:
Agreed, fixed.